src/argilla/client/users.py

```diff
@@ -0,0 +1,310 @@
+            - `User.from_id` to get an existing user from Argilla by its ID
+            - `User.list` to list all the users in Argilla
+
+        Args:
+            name: the name of the user to be managed. Defaults to None.
+            id: the ID of the user to be managed. Defaults to None.
+
+        Raises:
+            Exception: if the `User` is initialized via the `__init__` method.
+        """
+        error_msg = (
+            "`User` cannot be initialized via the `__init__` method, as it needs"
+            " to send requests to the Argilla API, so it can just be instantiated via"
+            " the following classmethods: `User.create`, `User.from_name`,"
+            " `User.from_id` and `User.list`; note that it must exist in"
+            " advance in Argilla, unless you are creating a new one."
+        )
+        if id is not None:
+            error_msg += f" As the `id` argument is not None, you should use `User.from_id('{id}')` instead."
+        if name is not None:
+            error_msg += f" As the `name` argument is not None, you should use `User.from_name('{name}')` instead."
+        raise Exception(error_msg)
+
+    def __str__(self) -> str:
+        return (
+            f"User(id={self.id}, username={self.username}, role={self.role},"
+            f" workspaces={self.workspaces}, api_key={self.api_key},"
+            f" first_name={self.first_name}, last_name={self.last_name}, role={self.role},"
+            f" inserted_at={self.inserted_at}, updated_at={self.updated_at})"
+        )
+
+    @staticmethod
+    def __active_client(httpx: bool = True) -> Union["httpx.Client", "AuthenticatedClient"]:
+        """Returns the active Argilla `httpx.Client` instance."""
+        try:
+            client = active_client().http_client
+            if httpx:
+                client = client.httpx
+            return client
+        except Exception as e:
+            raise RuntimeError(f"The `rg.active_client()` is not available or not respoding.") from e
+
+    def delete(self) -> None:
+        """Deletes the user from Argilla.
+
+        Raises:
+            BaseClientError: if the user cannot be deleted from Argilla.
+
+        Examples:
+            >>> from argilla import rg
+            >>> user = rg.User.from_name("my-user")
+            >>> user.delete()
+        """
+        try:
+            users_api.delete_user(self.__client, user_id=self.id)
+        except NotFoundApiError as e:
+            raise ValueError(
+                f"User with username=`{self.username}` doesn't exist in Argilla, so please"
+                " make sure that the name you provided is a valid one."
+            ) from e
+        except BaseClientError as e:
+            raise RuntimeError(f"Error while deleting user with username=`{self.username}` from Argilla.") from e
+
+    @classmethod
+    def __new_instance(cls, client: Optional["httpx.Client"] = None, ws: Optional["UserModel"] = None) -> "User":
```

Comment:
does `ws` stand for workspace user?

Reply:
Nope that was a copy-paste issue from the `__new_instance` method defined in `Workspace` 😄